.github/workflows/e2e-aws-custom.yml

```diff
@@ -0,0 +1,181 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: E2E (Custom)
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_or_branch:
+        description: 'pull request number or branch name'
+        required: true
+        default: 'main'
+      instance_type:
+        description: 'EC2 instance type'
+        required: true
+        default: 'g6.2xlarge'
+      e2e_script_args:
+        description: 'Arguments to pass to the e2e script (basic-workflow-tests.sh)'
+        required: true
+        default: '-m'
+      ilab_install_target:
+        description: 'pip install target for instructlab, e.g. . or .[vllm]'
+        required: true
+        default: '.'
+
+jobs:
+  start-runner:
+    name: Start external EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@fcfb31a5760dad1314a64a0e172b78ec6fc8a17e # v2.3.6
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-00c51d9c1374eda97
+          ec2-instance-type: ${{ github.event.inputs.instance_type }}
+          subnet-id: subnet-02d230cffd9385bd4
+          security-group-id: sg-06300447c4a5fbef3
+          iam-role-name: instructlab-ci-runner
+          aws-resource-tags: >
+            [
+              {"Key": "Name", "Value": "instructlab-ci-github-runner"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"},
+              {"Key": "GitHubWorkflow", "Value": "${{ github.workflow }}"},
+              {"Key": "GitHubRunID", "Value": "${{ github.run_id }}"}
+            ]
+
+  e2e:
+    name: E2E Test
+    needs: start-runner
+    runs-on: ${{ needs.start-runner.outputs.label }}
+
+    permissions:
+      pull-requests: write
```

Comment:
Can we add the warning message @russellb has in other workflows with this permission?

Reply:
nathan and I talked about this and he said its actually fine